examples/cms-batch/solve_spread/src/main.rs

```diff
@@ -0,0 +1,35 @@
+use std::io;
+
+fn main() {
+    let mut input = String::new();
+    io::stdin().read_line(&mut input).expect("Failed to read line");
+
+    let parts: Vec<&str> = input.trim().split_whitespace().collect();
+    if parts.len() != 2 {
+        eprintln!("Please provide exactly two numbers: N and K");
+        return;
+    }
+
+    let n: usize = match parts[0].parse() {
+        Ok(val) => val,
+        Err(_) => {
+            eprintln!("Invalid number for N");
+            return;
+        }
+    };
+
+    let k: usize = match parts[1].parse() {
+        Ok(val) => val,
+        Err(_) => {
+            eprintln!("Invalid number for K");
+            return;
+        }
+    };
```

Comment:
This could be a lot shorter, if you don't mind panicing:

```suggestion
    if parts.len() != 2 {
        panic!("Please provide exactly two numbers: N and K");
    }

    let n: usize = parts[0].parse().expect("Invalid number for N");
    let k: usize = parts[1].parse().expect("Invalid number for K");
```